### DISCUSSION TITLE
How to rotate characters to 90 degree ?

### DISCUSSION THREAD

**Discussion by murlai** [pymupdf/PyMuPDF#1951] [Looking for help]
How to rotate characters to 90 degree  before get **extractRAWDICT()** ?

only if rotated. if **lines["dir"]** not equal to **(0,0)**

<img width="685" alt="Screenshot 2022-10-08 at 10 18 09 AM" src="https://user-images.githubusercontent.com/10070890/194689510-4a4f4c98-1e84-42b4-a810-d691fce97344.png">
[7124453_PA_V1_R2.pdf](https://github.com/pymupdf/PyMuPDF/files/9738428/7124453_PA_V1_R2.pdf)

**Comment by JorjMcKie** [comment]
There is no way to do that. Every text extraction is based on the content of a `TextPage` object.

**Reply by murlai** [reply]
ooh. 

https://github.com/pymupdf/PyMuPDF/discussions/1950 In this discussion, I get some valuable information to get the exact font size. using the width and height of the text. 
It's working good when the rotations are 0°, 90°,180°,360°
First of all thanks ([JorjMcKie](https://github.com/JorjMcKie)) for this.
<img width="203" alt="Screenshot 2022-10-08 at 4 25 23 PM" src="https://user-images.githubusercontent.com/10070890/194703973-0c5b5b66-04aa-4c71-b264-0591239e1834.png">


but I face issues when the text is in some different rotations (like 55°,20°,42°,106°, etc) even though the arc-type or wrap texts.
 
<img width="203" alt="Screenshot 2022-10-08 at 4 20 11 PM" src="https://user-images.githubusercontent.com/10070890/194703807-c076add9-c918-49ee-8c6b-0186479911b2.png">
<img width="203" alt="Screenshot 2022-10-08 at 4 19 55 PM" src="https://user-images.githubusercontent.com/10070890/194703811-79d3a403-3c29-4b76-8fa0-0e685e9966ab.png">

**Reply by murlai** [reply]
I imagine a different way to get the exact font size. 
so I going to create a new pdf with straight every character based on the **lines["dir"]**
And then I going to calculate the exact font size. 

I don’t know whether it’s possible or not.

**Reply by JorjMcKie** [reply]
I recommend you look at the output of `mutool trace ...`. This is XML format, which also contains the `Tm` (="trm"). Try using an XML package like `lxml` to interpret it.
Here is the output for `horizontal_scaling.pdf` created by `mutool trace horizontal_scaling.pdf > trace.xml`.
[trace.zip](https://github.com/pymupdf/PyMuPDF/files/9739059/trace.zip)
